### PR TITLE
[receiver/kafkareceiver] enable errcheck

### PR DESCRIPTION
**Pull request by my-git9**
**Description:** <Describe what has changed.>

[receiver/kafkareceiver] enable errcheck

**Link to tracking Issue:** <Issue number if applicable>
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/9749

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: receiver/kafkareceiver/kafka_receiver.go [hunk 1]
```diff
@@ -140,7 +140,11 @@ func (c *kafkaTracesConsumer) Start(context.Context, component.Host) error {
 		autocommitEnabled: c.autocommitEnabled,
 		messageMarking:    c.messageMarking,
 	}
-	go c.consumeLoop(ctx, consumerGroup) // nolint:errcheck
+	go func() {
+		if err := c.consumeLoop(ctx, consumerGroup); err != nil {
+			return
+		}
+	}()
```
Comment (bogdandrutu) [inline receiver/kafkareceiver/kafka_receiver.go:147]: Looks like this is a fatal errors, we should report it to the host.

**Comment by mx-psi** [comment]
@my-git9 could you address https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/14596#discussion_r983725176 ? You will need to use the [`ReportFatalError` method from the `component.Host` interface](https://pkg.go.dev/go.opentelemetry.io/collector@v0.62.1/component#Host)

**Comment by dmitryax** [comment]
@my-git9 PTAL at the failing tests

**Comment by my-git9** [comment]
> @my-git9 PTAL at the failing tests

Sorry for taking so long to address this

**Review by mx-psi — APPROVED** [review]
LGTM, @bogdandrutu can you take another look?

**Review by codeboten — APPROVED** [review]

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**